src/engine/enginebuffer.cpp

```diff
@@ -1136,9 +1151,10 @@ void EngineBuffer::processSlip(int iBufferSize) {
     bool enabled = static_cast<bool>(load_atomic(m_slipEnabled));
     if (enabled != m_bSlipEnabledProcessing) {
         m_bSlipEnabledProcessing = enabled;
-        if (enabled) {
+        if (enabled || m_slipCancelled.fetchAndStoreAcquire(0)) {
             m_dSlipPosition = m_filepos_play;
             m_dSlipRate = m_rate_old;
+            m_slipCancelled = 0;
```

Comment:
>This is redundant now.

How so? Without this line, wouldn't slip_cancel need to be set to 0 immediately after setting it to 1?